refreshable/refreshable_validating.go

```diff
@@ -41,6 +27,22 @@ func (v *ValidatingRefreshable) LastValidateErr() error {
 // validatingFn successfully. This returns an error if the current value of the passed in Refreshable does not pass the
 // validatingFn or if the validatingFn or Refreshable are nil.
 func NewValidatingRefreshable(origRefreshable Refreshable, validatingFn func(interface{}) error) (*ValidatingRefreshable, error) {
+	mappingFn := func(i interface{}) (interface{}, error) {
+		if err := validatingFn(i); err != nil {
+			return nil, err
+		}
+		return nil, nil
+	}
+	return newValidatingRefreshable(origRefreshable, mappingFn, false)
+}
+
+// MapValidatingRefreshable is similar to NewValidatingRefreshable but allows for the function to return a mapping/mutation
+// of the input object in addition to returning an error. It must always return a value of same type.
```

Comment:
The last sentence is not actually true and we don't enforce it. For example in the test you added it goes from `string -> *url.URL`

Reply:
Yeah, this is supposed to mean that the returned `interface{}` has to always be the same type so the DefaultRefreshable will accept it. I tried to tweak the language, lmk if it makes sense

Reply:
ahh got it, that makes more sense. So basically the mapping function must be consistent with the returned type. For sake of example, this would be invalid as the type is changed if the default original is updated to anything but 1.
```go
r := refreshable.NewDefaultRefreshable(1)
vr, err := refreshable.MapValidatingRefreshable(r, func(i interface{}) (interface{}, error) {
		switch v := i.(int); v {
		case 1:
			return "1", nil
		default:
			return 2, nil
		}
	})
```